code/game/objects/items/deployable/barricade.dm

```diff
@@ -0,0 +1,204 @@
+
+#define METAL 1
+#define WOOD 2
+#define SAND 3
+
+//Largely utilized subtype, add some more types of barricades!
+/obj/item/deployable/barricade
+	name = "Generic Barricade"
+	desc = "You should never see this"
+	time_to_deploy = 3 SECONDS
+
+/obj/item/deployable/barricade/security
+	name = "security barricade"
+	desc = "A very sturdy barricade for use by Nanotrasen security personnel."
+	icon = 'icons/obj/objects.dmi'
+	icon_state = "barrier0"
+	deployed_object = /obj/structure/barricade/security
+	time_to_deploy = 3 SECONDS
+	w_class = WEIGHT_CLASS_SMALL
+
+/obj/item/storage/box/sec_barricades
+	name = "box of barricades"
+	desc = "A box full of sturdy security barricades"
+	icon_state = "secbox"
+	illustration = "syndiesuit"
+
+/obj/item/storage/box/sec_barricades/PopulateContents()
+	for(var/i in 1 to 7)
+		new /obj/item/deployable/barricade/security(src)
+
+//Barricades in structure form
+/obj/structure/barricade
+	name = "chest high wall"
+	desc = "Looks like this would make good cover."
+	anchored = TRUE
+	density = TRUE
+	max_integrity = 100
+	var/proj_pass_rate = 50 //How many projectiles will pass the cover. Lower means stronger cover
+	var/bar_material = METAL
+	var/pickup_delay = 8 SECONDS
```

Comment:
Why have this variable on the supertype, if pickup behaviour is added to the subtypes. Shouldn't pickup behaviour just be on this type?

Reply:
Yes it absolutely should

This is probably a mistake caused when I fluctuated between fully refactoring wooden and sandbag barricades too.

Reply:
This is resolved in the new PR now.